Initial commit. Added a new property to enable logging of client-side…

paragchandra opened the pull request:
… test exclusions. Copy value of 'forkEvery' parameter from original unit test task.

Commit:
Initial commit. Added a new property to enable logging of client-side test exclusions. Copy value of 'forkEvery' parameter from original unit test task.

jguerra commented on hydra-gradle-core/src/main/java/com/pandora/hydra/LazyTestExcluder.java:
```diff
@@ -43,13 +47,35 @@ public boolean isSatisfiedBy(FileTreeElement fileTreeElement) {
+            return;
+        }
+
+        String hostname = System.getenv("NODE_NAME");
```
Could you change this to be based off the slaveName in HydraPluginExtension instead?

paragchandra replied:
I think I tried that at first. The problem is slaveName is often explicitly unspecified on the HydraPluginExtension, so it comes back null. Or at least, because _I'm_ not specifying it there, it's coming back null. If that's unexpected, I can try it again.

jguerra replied:
Oh ok, perhaps then just provide a fallback in case NODE_NAME isn't defined (assuming NODE_NAME is jenkins specific)

jguerra commented on hydra-gradle-core/src/main/java/com/pandora/hydra/LazyTestExcluder.java:
```diff
@@ -21,12 +23,14 @@
     private volatile Set<String> blacklist;
 
-    public LazyTestExcluder(String projectName, Supplier<HydraClient> hydraClientSupplier) {
+    public LazyTestExcluder(String projectName, Supplier<HydraClient> hydraClientSupplier, Project project) {
```
We probably don't need to pass in the projectName now that the project itself is being passed in

paragchandra replied:
Yep, good point. I'll make the change.

Commit:
No need to pass in 'projectName' separately since we're already passing in 'project'.
First use the hydraExtension.slaveName if defined; then fall back to System.env('NODE_NAME')

jguerra reviewed (approved):
LGTM
